src/cmap/commands.ts

```diff
@@ -709,3 +715,54 @@ export class BinMsg {
     return { utf8: { writeErrors: false } };
   }
 }
+
+const MESSAGE_HEADER_SIZE = 16;
+const COMPRESSION_DETAILS_SIZE = 9; // originalOpcode + uncompressedSize, compressorID
+
+/**
+ * @internal
+ *
+ * An OP_COMPRESSED request wraps either an OP_QUERY or OP_MSG message.
+ */
+export class OpCompressedRequest {
```

Comment:
I like this abstraction but would be happy to remove it.  I think it makes sense conceptually - an OP_COMPRESSED wraps either an OP_QUERY or an OP_MSG.  However, our message _parsing_ logic is structured differently.  We first decompress a response into an OP_MSG or OP_QUERY, and then later we parse the response.  consequently, it's not as straightforward to create an OpCompresedResponse with the current structure.   If we'd prefer consistency, I can get rid of this class.